### PR TITLE
Add a FUSE file system implementation for Reliance.

### PR DESCRIPTION
**Pull request by jcdubois**
This patch is on top of the previous linux patch.

It adds a FUSE file system implementation for reliance.

One mounted the user can access the reliance file system like a normal file system through the selected mount point. You can interract with the reliance file system with the usual linux tools (mkdir, cp, rm, ln, vi, ...).

Note: As Reliance does not support access permission all users have the same access right on all files/directories.

### CODE REVIEW

File: tests/util/math.c [hunk 1]
```diff
@@ -480,8 +480,8 @@ uint64_t RedUint64DivMod32(
     However, it does so without ever actually dividing/modulating a 64-bit
     value, since such operations are not allowed in all environments.
 
-    @param ullDividend   The value to divide.
-    @param ullDivisor    The value to divide by.
+    @param ullDividend  The value to divide.
+    @param ullDivisor   The value to divide by.
     @param pullRemainder Populated with the remainder; may be NULL.
```
Comment (aamcampbell) [inline tests/util/math.c]: Oops--I missed this change when merging pull request 5 because I was ignoring whitespace changes.

**Comment by aamcampbell** [comment]
This is excellent, thanks for contributing! We recently discussed the need for a fuse implementation, so we appreciate your help with this.

Just so you know, we have the following steps to complete internally before merging this into the repository:

- Move reliance_fuse.c to os/linux/tools and merge os/linux/fuse.mk into os/linux/host.mk, so that the fuse implementation is built automatically with the Linux host tools.
- Edit reliance_fuse.c to match Datalight code style and naming conventions ([doc/coding_style.txt](https://github.com/datalightinc/reliance-edge/blob/master/doc/coding_style.txt))
- Try the implementation with a Linux filesystem test (such as [fsstress](https://github.com/linux-test-project/ltp/tree/master/testcases/kernel/fs/fsstress) or an internal Datalight test) to ensure it functions as expected.

I expect these steps to be this complete by the end of the week.

**Comment by jcdubois** [comment]
Please note that the fuse filesystem build is obviously dependent on the presence of the "libfuse-dev" package on the system.
I put it in its own directory to avoid having impact on the other tools build.

**Comment by aamcampbell** [comment]
Good thought. However, the fuse project should be included all projects (including those listed in projects/freertos/atmel and projects/freertos/st), and we want to avoid unnecessary duplication. One solution would be to use use "reliance_fuse" as a separate make target in os/linux/host.mk, so that `make` builds the existing host tools and `make reliance_fuse` builds the fuse implementation.

**Comment by jcdubois** [comment]
Note also that this preliminary implementation is very crude. It doesn't benefit from the transactional feature or reliance-edge. This could/should be improved by using the fuse_file_info structure in open/read/write/close implementation.

**Comment by jcdubois** [comment]
So, I added the less crude implementation.

**Comment by aamcampbell** [comment]
I did notice how the fuse_file_info structure was not used before. Using it should provide several benefits, including performance.

Unfortunately we had to re-implement the fuse port to avoid using GPL code by other authors (based on the copyright notice at the top of reliance_fuse.c), but that process has gone smoothly using another Datalight fuse implementation as a template and using code from your implementation that is specific to Reliance Edge. I hope to push it to GitHub today.

**Comment by aamcampbell** [comment]
I haven't had time to extensively test it yet, but I've pushed the FUSE implementation to GitHub with commit 27031645.

**Comment by jcdubois** [comment]
I had to remove the flag_nopath initialization as it was not supported on Ubuntu precise (LTS) which is the default build env for Travis.

Anyway, if not initialized, they would default to 0 which it the expected value here.

**Comment by aamcampbell** [comment]
I merged in this pull request, and put the changes through our automated build system with commit 13db89b6. This is a new way of merging pull requests for us that should work well for minor changes on code that does not require MISRA compliance. We still had to manually merge changes in our internal repositories, which is why a whitespace fix in os/linux/services/osbdev.c [got reverted](https://github.com/datalightinc/reliance-edge/commit/13db89b66945a9c6b10f811dc388ac6fc9bae140#diff-c457e0c8ea158f859e5620730e8a2272) by commit 13db89b6 (I missed it until now, so I'll fix it again in the next commit). I hope this makes sense to you.

**Pull request merged**